### PR TITLE
Add initial safechef:FoodSafetyCertificationCredential

### PR DESCRIPTION
**Pull request by BigBlueHat**
I can provide the SVG for the `credential.image` and `credential.issuer.image`. Just wasn't sure if you wanted them in this repo or elsewhere.

Hope this helps!
🎩

### CODE REVIEW

File: safechef/safechef-v1.context.jsonld [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "@context": {
+    "@version": 1.1,
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:3]:
```suggestion
```

No need for `@version` anymore.

File: safechef/safechef-v1.context.jsonld [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "@protected": true,
+
+    "safechef": "https://credential-handler.github.io/vc-examples/safechef/#",
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:6]:
```suggestion
    "safechef": "https://credential-handler.github.io/vc-examples/safechef/#",
```

We try not to use CURIEs in context files these days, just full URLs everywhere.

File: safechef/safechef-v1.context.jsonld [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    "name": "http://schema.org/name",
+    "description": "http://schema.org/description",
+    "image": {
+      "@id": "http://schema.org/image",
+      "@type": "@id"
+    },
+
+    "certificateNo": {
+      "@id": "safechef:certificateNo",
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:18]:
```suggestion
      "@id": "https://credential-handler.github.io/vc-examples/safechef/#certificateId",
```

File: safechef/safechef-v1.context.jsonld [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    "FoodSafetyCertificationCredential": "safechef:FoodSafetyCertificationCredential",
+
+    "name": "http://schema.org/name",
+    "description": "http://schema.org/description",
+    "image": {
+      "@id": "http://schema.org/image",
+      "@type": "@id"
+    },
+
+    "certificateNo": {
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:17]:
```suggestion
    "certificateId": {
```

File: safechef/safechef-v1.context.jsonld [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
+  "@context": {
+    "@version": 1.1,
+    "@protected": true,
+
+    "safechef": "https://credential-handler.github.io/vc-examples/safechef/#",
+
+    "FoodSafetyCertificationCredential": "safechef:FoodSafetyCertificationCredential",
```
[the remark this answers is not in the document]
Comment (msporny) [reply safechef/safechef-v1.context.jsonld:8]: We should make this a type-scoped context.

File: safechef/safechef.jsonld [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  "certificateNo": "8460c404-c0a5-4653-aeae-44d03a1a7334",
+  "examDate": "2011-11-11",
+  "testCode": "df52b514-e13a-42b7-8030-3f928a27d167"
```
Comment (davidlehn) [inline safechef/safechef.jsonld:22]: I'm not sure of the exact modeling for a certificate like this, but these certificate properties should be somewhere under `credentialSubject`.  There may be best practices to align with in VC-EDU or OpenBadges.  I'm not sure what the latest are though.

Food for thought:
https://w3c-ccg.github.io/vc-ed-models/#approach-2-open-badge-as-verifiable-credentials
They refer to 
https://imsglobal.github.io/openbadges-specification/ob_v3p0.html
Which has some `Certificate` references to the quite complex looking CTDL:
https://credreg.net/ctdl/terms

So many schemas... hard to know what to do!

- Properties for certificate itself should be under a property under `credentialSubject`.
- `issuanceDate` should be added to VC.
- Add an `id` under `credentialSubject` for jane doe, if possible.

File: safechef/safechef-v1.context.jsonld [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      "@id": "http://schema.org/image",
+      "@type": "@id"
+    },
+
+    "certificateNo": {
+      "@id": "safechef:certificateNo",
+      "@type": "xsd:string"
+    },
+    "examDate": {
+      "@id": "safechef:examDate",
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:22]:
```suggestion
      "@id": "https://credential-handler.github.io/vc-examples/safechef/#examDate",
```

File: safechef/safechef-v1.context.jsonld [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+      "@type": "xsd:dateTime"
+    },
+    "certificateNo": {
+      "@id": "safechef:certificateNo",
+      "@type": "xsd:string"
+    }
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:28]: Is this a duplicate of the above?

File: safechef/safechef-v1.context.jsonld [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+    "certificateNo": {
+      "@id": "safechef:certificateNo",
+      "@type": "xsd:string"
+    },
+    "examDate": {
+      "@id": "safechef:examDate",
+      "@type": "xsd:dateTime"
+    },
+    "certificateNo": {
```
Comment (davidlehn) [inline safechef/safechef-v1.context.jsonld:25]: Dup?  Was this supposed to be `testCode`?

File: safechef/safechef-v1.context.jsonld (hunk 1 above)
Comment (BigBlueHat) [reply safechef/safechef-v1.context.jsonld:3]: No? Standard is now 1.1 for VC processing?

File: safechef/safechef-v1.context.jsonld (hunk 2 above)
Comment (BigBlueHat) [reply safechef/safechef-v1.context.jsonld:6]: Because?

File: safechef/safechef.jsonld (hunk 6 above)
Comment (msporny) [inline safechef/safechef.jsonld:22]: We might want to look at Open Badges v3 and see how they would model this.

File: safechef/safechef-v1.context.jsonld (hunk 9 above)
Comment (BigBlueHat) [reply safechef/safechef-v1.context.jsonld:25]: Yeah. I think so. I'll correct it.

File: safechef/safechef.jsonld (hunk 6 above)
Comment (BigBlueHat) [reply safechef/safechef.jsonld:22]: K. Happy to take a look.

**Comment by BigBlueHat** [comment]
@msporny @davidlehn I still need to dig through the Open Badges data modelling approach some more, but this at least fixes the other (smaller) issues.

File: safechef/safechef-v1.context.jsonld [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+      "@id": "https://credential-handler.github.io/vc-examples/safechef/#FoodSafetyCertificationCredential",
+      "@context": {
+        "name": "http://schema.org/name",
+        "description": "http://schema.org/description",
+        "image": {
+          "@id": "http://schema.org/image",
+          "@type": "@id"
+        },
+
+        "certificateNo": {
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:15]:
```suggestion
        "certificateId": {
```

File: safechef/safechef-v1.context.jsonld [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+          "@id": "https://credential-handler.github.io/vc-examples/safechef/#certificateNo",
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:16]:
```suggestion
          "@id": "https://credential-handler.github.io/vc-examples/safechef/#certificateId",
```

File: safechef/safechef.jsonld (hunk 6 above)
Comment (msporny) [reply safechef/safechef.jsonld:22]: /cc @kayaelle @dmitrizagidulin thoughts? Is there a "Workforce Skills" example in OBv3 that we could re-use for this example credential?

**Comment by BigBlueHat** [comment]
@msporny @davidlehn feel free to let me know what else this one needs. 😉

File: safechef/safechef-v1.context.jsonld [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+  "@context": {
+    "@protected": true,
+
+    "FoodSafetyCertificationCredential": {
+      "@id": "https://credential-handler.github.io/vc-examples/safechef/#FoodSafetyCertificationCredential"
+    },
+
+    "name": "http://schema.org/name",
+    "description": "http://schema.org/description",
+    "image": "http://schema.org/image",
```
Comment (dlongley) [inline safechef/safechef-v1.context.jsonld:11]: When are we switching to the https versions of these?

Comment (BigBlueHat) [reply safechef/safechef-v1.context.jsonld:11]: Oh. Good question. Looks like they've made the transition in their examples/docs now, so I will too. Good call.

**Comment by BigBlueHat** [comment]
@dlongley made those tweaks. Let me know if there's anything else on this one. If not, I'll move on to others.

**Review by dlongley — APPROVED** [review]
Thanks! There are a couple of comments from Manu to look at as well. Note that if you click the little "refresh arrow circle thingy" next to the reviewer's name it will retrigger a review from them and put this PR on their list of PRs needing a review. A decent number of people just keep up to date with their own lists like that since they have so many other github notifications coming in all the time.

File: safechef/safechef-v1.context.jsonld [hunk 13]
```diff
@@ -0,0 +1,38 @@
+{
+  "@context": {
+    "@protected": true,
+
+    "FoodSafetyCertificationCredential": {
+      "@id": "https://credential-handler.github.io/vc-examples/safechef/#FoodSafetyCertificationCredential"
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:6]:
```suggestion
      "@id": "https://credential-handler.github.io/vc-examples/safechef#FoodSafetyCertificationCredential"
```

File: safechef/safechef-v1.context.jsonld [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+    },
+
+    "name": "https://schema.org/name",
+    "description": "https://schema.org/description",
+    "image": "https://schema.org/image",
+    "certification": {
+      "@id": "https://credential-handler.github.io/vc-examples/safechef/#certification"
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:13]:
```suggestion
      "@id": "https://credential-handler.github.io/vc-examples/safechef#certification"
```

File: safechef/safechef-v1.context.jsonld [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+    },
+
+    "FoodSafetyCertification": {
+      "@id": "https://credential-handler.github.io/vc-examples/safechef/#FoodSafetyCertification",
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:17]:
```suggestion
      "@id": "https://credential-handler.github.io/vc-examples/safechef#FoodSafetyCertification",
```

File: safechef/safechef-v1.context.jsonld [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+        "@protected": true,
+        "id": "@id",
+        "type": "@type",
+
+        "certificateNo": {
+          "@id": "https://credential-handler.github.io/vc-examples/safechef/#certificateNo",
+          "@type": "xsd:string"
+        },
+        "examDate": {
+          "@id": "https://credential-handler.github.io/vc-examples/safechef/#examDate",
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:28]:
```suggestion
          "@id": "https://credential-handler.github.io/vc-examples/safechef#examDate",
```

File: safechef/safechef-v1.context.jsonld [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+          "@type": "xsd:dateTime"
+        },
+        "testCode": {
+          "@id": "https://credential-handler.github.io/vc-examples/safechef/#testCode",
```
Comment (msporny) [inline safechef/safechef-v1.context.jsonld:32]:
```suggestion
          "@id": "https://credential-handler.github.io/vc-examples/safechef#testCode",
```

File: safechef/safechef.jsonld [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+  "type": [
+    "VerifiableCredential",
+    "FoodSafetyCertificationCredential"
+  ],
+  "description": "This individual has passed the SafeChef Food Safety Certification examination successfully.",
+  "credentialSubject": {
+    "id": "did:example:...janes-did...",
+    "name": "Jane Doe",
+    "certification": {
+      "@type": "FoodSafetyCertification",
```
Comment (msporny) [inline safechef/safechef.jsonld:15]:
```suggestion
      "type": "FoodSafetyCertification",
```

File: safechef/safechef.jsonld [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+      "certificateNo": "8460c404-c0a5-4653-aeae-44d03a1a7334",
```
Comment (msporny) [inline safechef/safechef.jsonld:16]:
```suggestion
      "certificateId": "urn:uuid:8460c404-c0a5-4653-aeae-44d03a1a7334",
```

File: safechef/safechef.jsonld [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+      "examDate": "2011-11-11",
+      "testCode": "df52b514-e13a-42b7-8030-3f928a27d167"
```
Comment (msporny) [inline safechef/safechef.jsonld:18]:
```suggestion
      "testCode": "urn:uuid:df52b514-e13a-42b7-8030-3f928a27d167"
```

**Review by msporny — CHANGES_REQUESTED** [review]
Still doesn't seem aligned w/ Open Badges? Some URL fixes.

**Comment by BigBlueHat** [comment]
> If we can use Open Badges v3 (completely), instead of creating our own JSON-LD Context, we should do that.

Definitely down with using anything/everything preexisting vs. reinventing. The Open Badges stuff I'd found earlier was out of date (apparently) as it lacked everything you referenced @msporny and the new stuff has all (apparently) been moved in the last two days to the 1EdTech org here on GitHub, so the spec is now at: https://1edtech.github.io/openbadges-specification/ob_v3p0.html

Now that I've found all the correct reference points, I'll rework with this to match their context and get rid of the custom stuff I'd built off of the "print" example.

**Review by msporny — APPROVED** [review]

**Comment by JSAssassin** [comment]
@davidlehn @msporny @gannan08 This PR was merged but doesn't follow the new contribution guidelines for the repo. Should `safechef` be added to the `credentials` dir? Will this be used by Chapi Playground?

**Comment by msporny** [comment]
> @davidlehn @msporny @gannan08 This PR was merged but doesn't follow the new contribution [guidelines](https://github.com/credential-handler/vc-examples#usage) for the repo. Should `safechef` be added to the `credentials` dir? Will this be used by Chapi Playground?

Yes (after it's been fixed up to align w/ the new formats), and yes.

**Pull request merged**